### PR TITLE
fix for #53

### PR DESCRIPTION
**Pull request by petschki**
and some features:

- uninstall profile
- CSRF warning when editing taxonomy data the first time (write on read)
- update/fix german translations

### CODE REVIEW

**Comment by petschki** [comment]
what's the reason why travis doesn't start?

File: src/collective/taxonomy/jsonimpl.py [hunk 1]
```diff
@@ -24,7 +29,7 @@ def __init__(self, context, request):
         utility_name = request.get('taxonomy', '')
         taxonomy = queryUtility(ITaxonomy, name=utility_name)
         if not taxonomy:
-            raise ValueError('Taxonomy `%s` could not be found.' % utility_name)
+            raise ValueError('Taxonomy `%s` could not be found.' % utility_name)  # noqa
```
Comment (hvelarde) [inline src/collective/taxonomy/jsonimpl.py]: please don't use generic `noqa` comments; it's better to explicitly specify what are you trying to ignore.

File: src/collective/taxonomy/exportimport.py [hunk 2]
```diff
@@ -62,13 +64,14 @@ def importTaxonomy(context):
...
-                        result[name] = unicode(config.get('taxonomy', name), 'utf-8')
+                        result[name] = unicode(
+                            config.get('taxonomy', name), 'utf-8')
                     except ConfigParser.NoOptionError:
                         pass
 
                 for name in ['is_single_select', 'is_required']:
                     try:
-                        result[name] = config.get('taxonomy', name) == 'true' and True
+                        result[name] = config.get('taxonomy', name) == 'true' and True  # noqa: E501
```
Comment (malthe) [inline src/collective/taxonomy/exportimport.py]: The `and True` here seems to be reminiscent of a time when there wasn't an explicit comparison involved – as far as I can tell it's unnecessary.

File: src/collective/taxonomy/jsonimpl.py [hunk 3]
```diff
@@ -47,6 +52,7 @@ def generate_json(self, root):
 
     def get_data(self):
         """Get json data."""
+        alsoProvides(self.request, IDisableCSRFProtection)
```
Comment (malthe) [inline src/collective/taxonomy/jsonimpl.py]: Why is this necessary here? No ZODB object seems to be modified in this code.

Comment (petschki) [reply src/collective/taxonomy/jsonimpl.py]: I have do dig here further, but if you have a clean Plone-5.1 installation, create a taxonomy and want to edit data for the first time, you get a CSRF warning from plone.protect ...

File: src/collective/taxonomy/exportimport.py (hunk 2 above)
Comment (petschki) [reply src/collective/taxonomy/exportimport.py]: you're right ... I'll change that

File: src/collective/taxonomy/jsonimpl.py (hunk 3 above)
Comment (malthe) [reply src/collective/taxonomy/jsonimpl.py]: Maybe it happens in the `_fixup` method.

You could argue that there shouldn't be any fixing upping on a `GET` request.

Comment (petschki) [reply src/collective/taxonomy/jsonimpl.py]: thanks for the pointer ... the `_fixup` was indeed the problem ... I've refactored this to get rid of CSRF warnings ...

Comment (malthe) [reply src/collective/taxonomy/jsonimpl.py]: The dilemma is this:

There's an old version of the data format and there is an upgrade step which migrates this to the new format by calling `_fixup()`.

But what if people try to use their site without – yet – having gone through the upgrade step.

I think that was the reasoning behind these fixing upper calls. And there used to be no upgrade step!

But now, perhaps the right move is simply to throw and error and log a warning "Data not upgraded" – ?

File: src/collective/taxonomy/utility.py [hunk 4]
```diff
@@ -302,3 +310,16 @@ def translate(self, msgid, mapping=None, context=None,
             pretty_path = pretty_path.rsplit(PRETTY_PATH_SEPARATOR, 1)[-1]
 
         return pretty_path
+
+    def _fixup(self):
+        # due to compatibility reasons this method fixes data structure
+        # for old Taxonomy instances.
+        # XXX: remove this in version 2.0 to prevent write on read
+        if self.order is None:
+            safeWrite(self, getRequest())
```
Comment (malthe) [inline src/collective/taxonomy/utility.py]: Great. Poor old `self.REQUEST` – acquisition has fallen out of favor...

Comment (petschki) [reply src/collective/taxonomy/utility.py]: strange .. I got an `AttributeError` when doing `self.REQUEST` in _fixup ?

Comment (malthe) [reply src/collective/taxonomy/utility.py]: I guess the utility is not acquisition-wrapped. Makes sense I guess.

Plone is really split brain in terms of acquisition and components.

**Comment by malthe** [comment]
I guess we're ready to merge this then?

**Comment by petschki** [comment]
yes ... and I also think we should release 1.5.0 ... changelog gets longer and longer 😉

**Comment by petschki** [comment]
@malthe but before I'd like to run the tests ... somehow my branch doesn't start automatically on travis-ci .... do you have an idea why?

**Comment by malthe** [comment]
```
While:
  Installing.
  Getting section download.
  Initializing section download.
  Installing recipe hexagonit.recipe.download.
  Getting distribution for 'hexagonit.recipe.download'.
Error: Wheels are not supported
The command "bin/buildout -N -q versions:setuptools= versions:zc.buildout=" failed and exited with 1 during .
```

**Comment by petschki** [comment]
I thought there would be some errors with this travis configuration, but this branch never ever gets started on travis-ci ... https://travis-ci.org/collective/collective.taxonomy says "This is not an active repository" .... I do not have privileges to activate it. Do you have @malthe ?

**Comment by malthe** [comment]
I don't really understand if I'm an admin or not. I can use "Settings" on that repo, but I also get the same error message.

**Comment by petschki** [comment]
just found this entry from june 29th stating some issues with de/activating repositories on https://www.traviscistatus.com/ ... I'll get in touch with travis support

**Comment by petschki** [comment]
ping @jensens are you able to activate this repository on travis-ci, or do you have an idea what's going wrong here? https://travis-ci.org/collective/collective.taxonomy
thx

**Comment by jensens** [comment]
I activated it on TravisCI (.org) - no idea why it was deactivated there.

**Comment by petschki** [comment]
@malthe finally
